### ISSUE TITLE
[mtl] fullscreen blocks waiting for next drawable if swapchain image count is 2

### ISSUE THREAD

**Issue by aleksijuvani** [gfx-rs/gfx#2900]
Fullscreen seems to block waiting for the next drawable for ridiculously long times (up to 60 milliseconds) if the swapchain image count (i.e. `maximumDrawableCount`) is set to 2 images instead of 3 images.

**Comment by kvark** [comment]
Is this still the case with the new API of #2882? Is this the case when using Metal directly?

**Comment by aleksijuvani** [comment]
Yes, this also happens with the new API of #2882. I'm not sure if this is something we're doing wrong in gfx-hal, or if this also happens when using Metal directly. I would have to put together a minimal repro using Metal directly, but I won't have time to do this for a while.

**Comment by aleksijuvani** [comment]
<img width="1392" alt="Screenshot 2019-07-12 at 17 41 55" src="https://user-images.githubusercontent.com/3168386/61136517-7c777d00-a4cc-11e9-841f-7a63729e7898.png">
This is also reproducible with the quad example and can be observed with the "Game Performance" profiler from Instruments.

**Comment by kvark** [comment]
Thanks for the info! It could be related to something we do with the swapchain on resize. We need to check Metal native examples, see how they handle resize and/or fullscreen, and make sure our implementation follows that (speaking of `build_swapchain` routine specifically). Maybe we end up holding the old drawables unnecessary, or something like this....

**Comment by mtak-** [comment]
I can repro this on macos, but on iOS I cannot. That should narrow things down a bit.
